### PR TITLE
Testing for RON files

### PR DESCRIPTION
**Pull request by peter-scholtens**
As a solution to the proposed test as mentioned in #192 I created a two functions: read_original()  and test_sequence(). Using an assert equal macro various files can be tested. Two examples are given: the first one fails as the mapping sequence changes, the secvond one is correct, as the map only contains one item.

### CODE REVIEW

File: tests/preserve_sequence.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+struct Nested {
+    a: String,
+    b: char,
+}
+
+fn read_original(file_name: &str) -> (String) {
+    let input_path = format!("{}/tests/{}", env!("CARGO_MANIFEST_DIR"), file_name);
```
Comment (kvark) [inline tests/preserve_sequence.rs]: Why don't we just include the original with `include_str!` or something?

File: tests/preserve_sequence.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    fs::read_to_string(input_path)
+        .expect("Something went wrong reading the file")
+}
+
+fn test_sequence(file_name: &str) -> (String) {
```
Comment (kvark) [inline tests/preserve_sequence.rs]: perhaps this should be called `test_roundtrip` instead?

File: tests/preserve_sequence.rs (hunk 1 above)
Comment (peter-scholtens) [reply tests/preserve_sequence.rs]: Didn't know it exists. However, if I try, the compiler keeps complaining "error: argument must be a string literal", unless I copy the string into the function. But then the flexibility of calling various files is gone.

File: tests/preserve_sequence.rs (hunk 2 above)
Comment (peter-scholtens) [reply tests/preserve_sequence.rs]: Good suggestion. I though about several names (reciprocal, symmetric etc) but couldn't find a brief and correct description. See 1272b61ce01ae321d3e0050bdf6b6d87b8ea61d1

Comment (peter-scholtens) [reply tests/preserve_sequence.rs]: Forgot to update calling functions, see update in e3ada7f0c2876a598b17888cab2f9e445b3c9632 Now the first example fails as the mapping order is modified, while the second example proofs the test is okay.

File: tests/preserve_sequence.rs (hunk 1 above)
Comment (kvark) [reply tests/preserve_sequence.rs]: What I was thinking:
```rust
fn read_original(source: &str) -> String {...}

fn test_xxx() {
  let file1 = include_str!("my_file1.ron");
  assert_eq!(
    read_original(file1),
    make_roundtrip(file1),
  );  
}
```

Comment (peter-scholtens) [reply tests/preserve_sequence.rs]: Ah, okay. Then the file is only read once, and processed twice. A much nicer way indeed. See 0fa9da321841aa6dea640007007348000a67f9a0

**Comment by peter-scholtens** [comment]
Like this ? c0d52d20cf1d2f65e29af51f797a5bf347ba317a

**Comment by peter-scholtens** [comment]
It should fail testing on the first example1 as the sequence is not preserved. Or should this be listed as a separate bug? The second test should pass succesfully.

**Comment by kvark** [comment]
thank you!
bors r+

**Pull request merged**